### PR TITLE
Use `typing.Self`

### LINKED ISSUE #240: `CacheRegion.configure` should return `typing_extensions.Self`
https://github.com/sqlalchemy/dogpile.cache/blob/96b8b2858f29c44893257b2487fd38554d4346ef/dogpile/cache/region.py#L429

Returning a fixed type here is unfriendly to subclassing `CacheRegion` to e.g. add new methods.

### PR DESCRIPTION
**Pull request by Viicos**
Fixes #240

### CODE REVIEW

File: dogpile/cache/api.py [hunk 1]
```diff
@@ -190,7 +195,7 @@ def __init__(self, arguments: BackendArguments):  # pragma NO COVERAGE
         raise NotImplementedError()
 
     @classmethod
-    def from_config_dict(cls, config_dict, prefix):
+    def from_config_dict(cls, config_dict: Mapping[str, Any], prefix: str) -> Self:
```
Comment (Viicos) [inline dogpile/cache/api.py:198]: Adding types here showed something interesting:

https://github.com/sqlalchemy/dogpile.cache/blob/fd9d31a72f9b8bef090f9496eacdfc77fc2ea0da/dogpile/cache/region.py#L497-L500

`_config_prefix` can be `None` according to the `__init__` signature, but `from_config_dict` requires a non-`None` `str` value

File: dogpile/cache/api.py [hunk 2]
```diff
@@ -8,6 +9,10 @@
 from typing import Optional
 from typing import Sequence
 from typing import Union
+if sys.version_info >= (3, 11):
```
Comment (zzzeek) [inline dogpile/cache/api.py:12]: can this conditional be moved to dogpile/util/compat.py , or a new module dogpile/util/typing.py?  then we import as "from ..util.typing import Self"  thanks

File: dogpile/cache/api.py (hunk 1 above)
Comment (zzzeek) [reply dogpile/cache/api.py:198]: it's assumed they would go together as both None or both present.   these are private variables so if typing is failing use `type: ignore` for now

File: dogpile/cache/proxy.py [hunk 3]
```diff
@@ -9,10 +9,14 @@
 .. versionadded:: 0.5.0  Added support for the :class:`.ProxyBackend` class.
 
 """
-
+import sys
 from typing import Mapping
 from typing import Optional
 from typing import Sequence
+if sys.version_info >= (3, 11):
```
Comment (zzzeek) [inline dogpile/cache/proxy.py:16]: import `from ..util.typing import Self`

File: dogpile/cache/region.py [hunk 4]
```diff
@@ -18,6 +19,10 @@
 from typing import Tuple
 from typing import Type
 from typing import Union
+if sys.version_info >= (3, 11):
```
Comment (zzzeek) [inline dogpile/cache/region.py:22]: import `from ..util.typing import Self`

**Review by zzzeek — CHANGES_REQUESTED** [review]
hi and thanks for this!  let's move all that sys.version_info conditional into just one place.  from there I would also add a changelog file but I can do that on this end.

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **zzzeek** to try to get revision 2bd4d45ac1847b961040ecc5978f3edf0299b5bc of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
New Gerrit review created for change 2bd4d45ac1847b961040ecc5978f3edf0299b5bc: https://gerrit.sqlalchemy.org/c/sqlalchemy/dogpile.cache/+/4739

File: dogpile/cache/api.py (hunk 2 above)
Comment (Viicos) [reply dogpile/cache/api.py:12]: I was thinking about doing this as well

**Comment by zzzeek** [comment]
beautiful, thanks

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **zzzeek** to try to get revision 33ea2b3d4222ad48d5a6b4493d45c9ee005ea160 of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
Patchset 33ea2b3d4222ad48d5a6b4493d45c9ee005ea160 added to existing Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/dogpile.cache/+/4739

**Comment by sqla-tester** [comment]
Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/dogpile.cache/+/4739 has been **merged**. Congratulations! :)

**Comment by zzzeek** [comment]
thank you, since we dont release regularly will release now

**Comment by zzzeek** [comment]
ok you're all set

**Pull request closed without merging**